### ISSUE TITLE
Register DSC client.ps1 

### ISSUE THREAD

**Issue by ArieHein** [AesonDev/DSC-COnfig#2]
Hi,

As for this, you have mixed two things: AgentID and ConfigurationID.
Although both are GUIDs they serve differnt purposes and act with different parts of the script.

ConfigurationID is the old V4 method for Nodes registering with the Pull server. You have to set it yourself and manage it. You can set couple of nodes to have the same ConfigrationID which allows you to pull same configuration scipt to create matching nodes.
ConfigurationID works with PartialConfigs in the LCM script.

RegistrationKey is the new V5 method for Nodes registration with the pull server. You dont have to set it or manage it. By using a shared "secret" - thats the Key you use in the LCM and in the txt file on the pull server, a node registers to the pull and the pull assigns it an AngentID which i unique to the each node.
RegistrationID works with ConfigurationsNames.

So basically ConfigurationNames is the way to do "Partial Configurations" when you use the new method, a.k.a. RegistrationKey.

If you want to use the new RegistraionKey you use the ConfigurationNames, theres no point to use the PartialConfiguration blocks, thats the old v4 way.
In this script you basically used the ConfigurationID so you done need the ConfiguratioNames in the LCM, thats why it seems to be "working without AgentID beening needed"

**Comment by AesonDev** [comment]
I see the same issue  here : 
https://windowsserver.uservoice.com/forums/301869-powershell/suggestions/12337116-multiple-configurationnames-results-in-failure-dur

I will post the code.

**Comment by AesonDev** [comment]
It is apparently impossible to use ConfigurationNames with multiple configuration...

**Comment by ArieHein** [comment]
If you use ConfigurationNames   = @{"BaseOs","IIS"}. You the need tei configuration scripts that will be called BaseOs.mof and IIS.mof on the pull server, and the the LCM will pick them both.
That's why I wrote that ConfigurationNames is the "PartialConfiguration" of v4.
 there's no need to have any Partial blocks in your LCM script nor files caled partialxxxxxx.mof waiting on the pull server.

**Comment by danielgatley** [comment]
Even with WMF 5 RTM this doesn't seem to work, as soon as multiple configurations are specified the pull fails.  If only one configuration name is specified the pull works, testing each configuration name individually work so it's not a problem with missing mof files.  This also occurs when the configuration names are specified on the pull server.
